src/executor/mod.rs

```diff
@@ -187,9 +188,14 @@ impl EvmExecutor {
                     }
                 }
             }
-            if (acc.is_empty() && !info.is_empty()) || acc.code_hash.0 != info.code_hash.0 {
-                acc_data.poseidon_code_hash = H256::from(info.code_hash.0);
-                acc_data.keccak_code_hash = H256::from(info.keccak_code_hash.0);
+            if (acc.is_empty() && !info.is_empty()) || acc.keccak_code_hash.0 != info.code_hash.0 {
+                assert_ne!(
+                    info.poseidon_code_hash,
+                    B256::ZERO,
+                    "poseidon_code_hash didn't update"
+                );
```

Comment:
It will also help to log a `error` here, with the `acc` details as well. However, you could simply add a `// TODO` here so that it can be included via the tracing API later.

Reply:
it panics instead of logging..

Reply:
let me add acc to the panic message